### ISSUE TITLE
[infra] Windows coverage hangs on Dart 3.4.0

### ISSUE THREAD

**Issue by dcharkes** [dart-lang/native#1154]
Failing run:

* https://github.com/dart-lang/native/actions/runs/9094025881/job/24994214663

Latest successful run:

* https://github.com/dart-lang/native/actions/runs/9002742091/job/24731730603

```
Run dart pub global run coverage:test_with_coverage
  dart pub global run coverage:test_with_coverage
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    DART_HOME: C:\hostedtoolcache\windows\dart\[3](https://github.com/dart-lang/native/actions/runs/9094025881/job/24994214663#step:33:3).4.0\x64
    PUB_CACHE: C:\Users\runneradmin/.pub-cache

[...]

✅ test\example\native_add_library_test.dart: native_add build

🎉 92 tests passed.
Error: The operation was canceled.
```

**Comment by dcharkes** [comment]
Reverting to the last commit that succeeded doesn't help:

* https://github.com/dart-lang/native/actions/runs/9110139872/job/25044523142

So, either the coverage package got updated, or the Windows GitHub bots.

**Comment by dcharkes** [comment]
Running on a Windows VM with a recent Dart seems to terminate:

```
C:\src\dart-lang\native\pkgs\native_assets_cli>dart test
Resolving dependencies in `C:\src\dart-lang\native\pkgs\native_assets_cli`...
Downloading packages...
Got dependencies in `C:\src\dart-lang\native\pkgs\native_assets_cli`.
Building package executable... (1.3s)
Built test:test.
00:13 +92: All tests passed!

C:\src\dart-lang\native\pkgs\native_assets_cli>dart --version
Dart SDK version: 3.5.0-160.0.dev (dev) (Wed May 15 17:06:58 2024 -0700) on "windows_x64"
```

**Comment by dcharkes** [comment]
GitHub image `windows-2019` also hangs:

* https://github.com/dart-lang/native/actions/runs/9113626982/job/25055608015

**Comment by dcharkes** [comment]
Reverting to Dart 3.3.0 stable makes the CI succeed.

* https://github.com/dart-lang/native/actions/runs/9113728428/job/25055967575

**Comment by dcharkes** [comment]
Update, this will only be fixed in 3.5.0:

* https://github.com/dart-lang/coverage/issues/490#issuecomment-2221630546
